src/main/java/org/folio/fqm/service/ResultSetService.java

```diff
@@ -1,40 +1,65 @@
 package org.folio.fqm.service;
 
+import org.folio.fqm.exception.EntityTypeNotFoundException;
+import org.folio.fqm.repository.EntityTypeRepository;
 import org.folio.fqm.repository.ResultSetRepository;
+import org.folio.fqm.utils.IdColumnUtils;
+import org.folio.querytool.domain.dto.EntityType;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
 
 import lombok.RequiredArgsConstructor;
 
+import java.util.ArrayList;
+import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
 import java.util.UUID;
+import java.util.concurrent.atomic.AtomicInteger;
 import java.util.function.Function;
 import java.util.stream.Collectors;
 
-import static org.folio.fqm.repository.EntityTypeRepository.ID_FIELD_NAME;
-
 @Service
 @RequiredArgsConstructor(onConstructor = @__(@Autowired))
 public class ResultSetService {
 
   private final ResultSetRepository resultSetRepository;
+  private final EntityTypeRepository entityTypeRepository;
 
   public List<Map<String, Object>> getResultSet(UUID entityTypeId,
                                                 List<String> fields,
-                                                List<UUID> ids) {
+                                                List<List<String>> ids) {
     List<Map<String, Object>> unsortedResults = resultSetRepository.getResultSet(entityTypeId, fields, ids);
     // Sort the contents in Java code as sorting in DB views run very slow intermittently
-    return getSortedContents(ids, unsortedResults);
+    return getSortedContents(entityTypeId, ids, unsortedResults);
   }
 
-  private static List<Map<String, Object>> getSortedContents(List<UUID> contentIds, List<Map<String, Object>> unsortedResults) {
-    Map<UUID, Map<String, Object>> contentsMap = unsortedResults.stream()
-      .collect(Collectors.toMap(content -> (UUID) content.get(ID_FIELD_NAME), Function.identity()));
-    return contentIds.stream()
+  private List<Map<String, Object>> getSortedContents(UUID entityTypeId, List<List<String>> contentIds, List<Map<String, Object>> unsortedResults) {
+    EntityType entityType = entityTypeRepository.getEntityTypeDefinition(entityTypeId)
+      .orElseThrow(() -> new EntityTypeNotFoundException(entityTypeId));
+    List<String> idColumnNames = IdColumnUtils.getIdColumnNames(entityType);
+    Map<List<String>, Map<String, Object>> contentsMap = unsortedResults.stream()
+      .collect(Collectors.toMap(content -> {
+            List<String> keys = new ArrayList<>();
+            idColumnNames.forEach(columnName -> keys.add(content.get(columnName).toString()));
```

Comment:
Do we need to worry about `content.get(columnName)` ever returning `null` here?

Reply:
The only way this would happen is if a record has a null value for one of its ID columns. So it could theoretically be a problem if we allow null values for ID columns, but I don't think it's something that would happen for our current entity types.

Reply:
if we want to allow nullable id columns, we can change this to something like:
```
var columnValue = content.get(columnName);
keys.add(columnValue == null ? null : columnValue.toString());
```